application/src/main/java/com/sanction/lightning/resources/FacebookResource.java

```diff
@@ -112,6 +117,60 @@ public Response getPhotos(@Auth Key key, @QueryParam("username") String username
   }
 
   /**
+   * Publishes to a users facebook.
+   * @param key The authentication key for the requesting application.
+   * @return Response that the photo either uploaded or failed.
+   */
+  @POST
+  @Path("/publish")
+  @Consumes(MediaType.MULTIPART_FORM_DATA)
+  public Response publish(@Auth Key key, @QueryParam("username") String username,
+                               @FormDataParam("file") final InputStream inputStream,
+                               @FormDataParam("file") final FormDataContentDisposition
+                                         contentDispositionHeader,
+                               @FormDataParam("message") final String message) {
+    if (username == null) {
+      return Response.status(Response.Status.BAD_REQUEST)
+              .entity("'username' query parameter is required for publicPhoto").build();
+    }
+
+    // Should get an empty string if the user didn't type anything
+    if (message == null) {
+      return Response.status(Response.Status.BAD_REQUEST)
+              .entity("'message' query parameter is required for publicPhoto").build();
+    }
+
+    if (inputStream == null) {
+      LOG.error("Bad InputStream");
+      return Response.status(Response.Status.INTERNAL_SERVER_ERROR)
```

Comment:
I think this might be better to be a Bad Request? It wasn't really an internal server error if the user didn't supply an InputStream.